### PR TITLE
React/DP-14339:  EmergencyHeader Component.

### PR DESCRIPTION
**Pull request by smurrayatwork**
## Description
- (React) [EmergencyHeader] DP-14339: Adds the EmergencyHeader component.

## Related Issue / Ticket

- [JIRA issue](https://jira.mass.gov/browse/DP-14339)

## Steps to Test

1. Pull the branch, start up storybook. Browse to molecules -> EmergencyHeader and verify the functionality works as intended.

## Screenshots
Screen Shot 2019-06-13 at 5 01 51 PM

### CODE REVIEW

File: react/src/components/molecules/EmergencyHeader/index.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        <span>{prefix}</span>
+      </span>
+      <span className="ma__emergency-header__title">{ title({ ...linkArgs }) }</span>
+    </h2>
+  );
+};
+
+EmergencyHeader.propTypes = {
+  /** A function whose return value is displayed in the text to the right of the divider bar. */
+  title: PropTypes.func.isRequired,
```
Comment (avrilpearl) [inline react/src/components/molecules/EmergencyHeader/index.js]: can you make it possible to pass a string or function here per standup conversation

File: react/src/components/molecules/EmergencyHeader/style.scss [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+@mixin ma-emergency-header ($color: $c-highlight) {
+  @if $color == $c-highlight {
+    color: $c-font-base;
+
+  } @else {
+    color: $c-font-inverse;
+  }
+  background-color: rgba($color, 1);
+  border-color: rgba($color, 1);
+  box-shadow: 0 0.25rem 0.5rem rgba(#000, 0.25);
```
Comment (avrilpearl) [inline react/src/components/molecules/EmergencyHeader/style.scss]: note: we should make this a mixin/variable in a follow up where used.

File: react/src/components/molecules/EmergencyHeader/style.scss [hunk 3]
```diff
@@ -0,0 +1,63 @@
+@import 'global';
+@import '00-base/mixins';
```
Comment (avrilpearl) [inline react/src/components/molecules/EmergencyHeader/style.scss]: this is imported in global so you can remove it here
```suggestion
```

File: react/src/components/molecules/EmergencyHeader/style.scss [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+@import '02-molecules/emergency-header';
+
+@mixin ma-emergency-header ($color: $c-highlight) {
```
Comment (avrilpearl) [inline react/src/components/molecules/EmergencyHeader/style.scss]: this should be moved into the assets file for this molecule that is imported above (lines 5 - 63)

Comment (clairesunstudio) [reply react/src/components/molecules/EmergencyHeader/style.scss]: maybe create a mixin file for this

File: react/src/components/molecules/EmergencyHeader/style.scss [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+  @if $color == $c-highlight {
+    color: $c-font-base;
+
+  } @else {
+    color: $c-font-inverse;
+  }
+  background-color: rgba($color, 1);
+  border-color: rgba($color, 1);
+  box-shadow: 0 0.25rem 0.5rem rgba(#000, 0.25);
```
Comment (clairesunstudio) [inline react/src/components/molecules/EmergencyHeader/style.scss]: I don't see a `box-shadow` on the patternlab side. I agree that it should have one but maybe should be added on the organism instead, including the alerts not on top of alerts
```suggestion
```

Comment (clairesunstudio) [reply react/src/components/molecules/EmergencyHeader/style.scss]: or if we are going to use this directly in budget, without the organism, then there should be a prop? what do you think @avrilpearl

File: react/src/components/molecules/EmergencyHeader/index.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  /** A string that controls different color themes for the component. */
+  theme: PropTypes.oneOf(['', 'c-highlight', 'c-primary-alt', 'c-primary', 'c-gray', 'c-error']),
```
Comment (clairesunstudio) [inline react/src/components/molecules/EmergencyHeader/index.js]:
```suggestion
  theme: PropTypes.oneOf(['c-highlight', 'c-primary-alt', 'c-primary', 'c-gray', 'c-error']),
```

File: react/src/components/molecules/EmergencyHeader/EmergencyHeader.stories.js [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+storiesOf('molecules', module)
+  .addDecorator(withKnobs({ escapeHTML: false }))
+  .add('EmergencyHeader', (() => {
+    const iconName = select('Icon: name', svgOptions, 'alert');
+    const titleText = text('title', 'The State is experiencing severe weather due to the winter storm Paula.');
+    const props = {
+      title: ({ theme, linkClasses }) => <a className={linkClasses} href="https://www.mass.gov">{titleText}</a>,
+      theme: select('theme', {
+        'c-highlight (default)': 'c-highlight',
+        'no theme': '',
```
Comment (clairesunstudio) [inline react/src/components/molecules/EmergencyHeader/EmergencyHeader.stories.js]:
```suggestion
```

File: react/src/components/molecules/EmergencyHeader/index.js [hunk 8]
```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import airbnbTypes from 'airbnb-prop-types';
+import Icon from '../../atoms/icons/Icon';
+
+import './style.css';
+
+const EmergencyHeader = (props) => {
+  const { title, icon, prefix } = props;
+  const theme = props.theme.length === 0 ? 'c-highlight' : props.theme;
```
Comment (clairesunstudio) [inline react/src/components/molecules/EmergencyHeader/index.js]: same comment as alert molecule, can we use classnames here

File: react/src/components/molecules/EmergencyHeader/style.scss (hunk 5 above)
Comment (avrilpearl) [reply react/src/components/molecules/EmergencyHeader/style.scss]: i think the organism as the show updates/emergency alert molecules are dependent on whether their are alerts or not

**Review by avrilpearl — APPROVED** [review]

**Comment by clairesunstudio** [comment]
Screen Shot 2019-06-17 at 3 17 58 PM
sorry just notice one more thing, when the prefix is not passed, the border should not be rendered

**Review by clairesunstudio — APPROVED** [review]

**Pull request merged**